projects/ng-components/src/lib/controls/advanced-search/advanced-search.component.ts

```diff
@@ -0,0 +1,81 @@
+
+@Component({
+  selector: 'lib-advanced-search',
+  templateUrl: './advanced-search.component.html'
+})
+export class AdvancedSearchComponent implements OnInit {
+  @Output() advancedSearchChanged: EventEmitter<FilterClause[]> = new EventEmitter<FilterClause[]>();
+  @Input('search-options') searchOptions: SearchOption[] = [];
+  @Input() filters?: FilterClause[];
+
+  public selectedField?: SearchOption;
+  public fieldValue?: string;
+  public fieldValueDateFrom: any;
+  public fieldValueDateTo: any;
+
+  constructor() { }
+
+  public ngOnInit(): void { }
+
+  public selectedFieldChanged() {
+    this.fieldValue = undefined;
+  }
+
+  public search() {
+    if (!this.filters) {
+      this.filters = [];
+    }
+    // the filter is of type 'daterange'
+    if (this.selectedField && this.selectedField.dataType && this.selectedField.dataType === 'daterange') {
+      if (!this.selectedField.multiTerm) {
+        this.filters = this.filters.filter((f) => {
+          return f.member !== QueryParameters.FILTER_FROM && f.member !== QueryParameters.FILTER_TO;
+        });
+      }
+      if (this.fieldValueDateFrom) {
+        const filterClauseFrom = new FilterClause(QueryParameters.FILTER_FROM, this.fieldValueDateFrom, Operators.GREATER_THAN_EQUAL.value as FilterClause.Op, 'datetime', this.searchOptions);
+        this.filters.push(filterClauseFrom);
+      }
+
+      if (this.fieldValueDateTo) {
+        const filterClauseTo = new FilterClause(QueryParameters.FILTER_TO, this.fieldValueDateTo, Operators.LESS_THAN_EQUAL.value as FilterClause.Op, 'datetime', this.searchOptions);
+        this.filters.push(filterClauseTo);
+      }
+    }
+    // the filter isn't of type 'daterange' (everything else)
+    if (this.selectedField && this.selectedField.dataType && this.selectedField.dataType !== 'daterange') {
+      if (this.fieldValue === undefined) { // handle empty field value
+        return;
+      }
+      // create the filterClause
+      const filterClause = new FilterClause(this.selectedField.field, this.fieldValue, Operators.EQUALS.value as FilterClause.Op, this.selectedField.dataType, this.searchOptions); // operator is always 'equals'...
+
+      if (!this.selectedField.multiTerm) { // multiTerm means that we can have multiple filter values of the SAME filter clause (eg "filter case types that are Phone OR Address")
+        this.filters = this.filters.filter((f) => {
+          return f.member !== this.selectedField!.field;
+        });
+      }
+
+      this.filters.push(filterClause);
+      this.fieldValue = undefined;
+    }
+    this.advancedSearchChanged.emit(this.filters);
+  }
+
+  public removeFilter(index: number) {
+    if (this.filters && this.filters.length >= index) {
+      this.filters.splice(index, 1);
+      this.advancedSearchChanged.emit(this.filters);
+    }
+  }
+
+  public clear() {
+    this.fieldValueDateFrom = undefined;
+    this.fieldValueDateTo = undefined;
+    this.filters = [];
```

Comment:
θες να κάνεις clear και το selectedField
```suggestion
    this.filters = [];
    this.selectedField = undefined;
```